### ISSUE TITLE
Gradients will be None loss=nan, my GPU is 1660ti(LAPTOP)

### ISSUE THREAD

**Issue by survival93** [bmaltais/kohya_ss#293]
D:\AI\Lora\kohya_ss\venv\lib\site-packages\torch\utils\checkpoint.py:25: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
  
  
What can I do to fix this

**Comment by xelmirage** [comment]
same problem, log:




Folder 100_a: 3600 steps
max_train_steps = 3600
stop_text_encoder_training = 0
lr_warmup_steps = 0
accelerate launch --num_cpu_threads_per_process=2 "train_network.py" --pretrained_model_name_or_path="F:/github/stable-diffusion-webui/models/Stable-diffusion/chilloutmix_NiPrunedFp32.safetensors" --train_data_dir="E:/lora/image" --reg_data_dir="E:/lora/Regularisation" --resolution=512,512 --output_dir="E:/lora/model" --logging_dir="E:/lora/logging" --network_alpha="1" --save_model_as=safetensors --network_module=networks.lora --text_encoder_lr=5e-5 --unet_lr=0.0001 --network_dim=8 --output_name="a" --lr_scheduler_num_cycles="1" --learning_rate="0.0001" --lr_scheduler="constant" --train_batch_size="1" --max_train_steps="3600" --save_every_n_epochs="1" --mixed_precision="fp16" --save_precision="fp16" --seed="1234" --caption_extension=".txt" --cache_latents --optimizer_type="AdaamW" --max_data_loader_n_workers="1" --clip_skip=2 --bucket_reso_steps=64 --mem_eff_attn --gradient_checkpointing --xformers --bucket_no_upscale
prepare tokenizer
Use DreamBooth method.
prepare images.
found directory E:\lora\image\100_a contains 36 image files
3600 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 1
  resolution: (512, 512)
  enable_bucket: False

  [Subset 0 of Dataset 0]
    image_dir: "E:\lora\image\100_a"
    image_count: 36
    num_repeats: 100
    shuffle_caption: False
    keep_tokens: 0
    caption_dropout_rate: 0.0
    caption_dropout_every_n_epoches: 0
    caption_tag_dropout_rate: 0.0
    color_aug: False
    flip_aug: False
    face_crop_aug_range: None
    random_crop: False
    is_reg: False
    class_tokens: a
    caption_extension: .txt


[Dataset 0]
loading image sizes.
100%|███████████████████████████████████████████████████████████████| 36/36 [00:00<00:00, 96.75it/s]
prepare dataset
prepare accelerator
Using accelerator 0.15.0 or above.
load StableDiffusion checkpoint
loading u-net: <All keys matched successfully>
loading vae: <All keys matched successfully>
loading text encoder: <All keys matched successfully>
Replace CrossAttention.forward to use FlashAttention (not xformers)
[Dataset 0]
caching latents.
100%|███████████████████████████████████████████████████████████████| 36/36 [00:05<00:00,  6.12it/s]
import network module: networks.lora
create LoRA network. base dim (rank): 8, alpha: 1.0
create LoRA for Text Encoder: 72 modules.
create LoRA for U-Net: 192 modules.
enable LoRA for text encoder
enable LoRA for U-Net
prepare optimizer, data loader etc.
use AdamW optimizer | {}
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 3600
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 3600
  num epochs / epoch数: 1
  batch size per device / バッチサイズ: 1
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 3600
steps:   0%|                                                               | 0/3600 [00:00<?, ?it/s]epoch 1/1
F:\github\kohya_ss\venv\lib\site-packages\torch\utils\checkpoint.py:25: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
steps:  31%|████████████▍                           | 1114/3600 [21:08<47:11,  1.14s/it, loss=0.129]

**Comment by tildebyte** [comment]
For posterity - this is #NotAnIssue, see https://github.com/kohya-ss/sd-scripts/issues/323, to wit:

>Grandient checkpointing seems to be enabled by specifying requires_grad_(True) for the first parameter of the model. However, ***LoRA training does not train that parameter***, so this warning seems to appear.

(above emphasis is mine)
